plugin/trino-oracle/src/main/java/io/trino/plugin/oracle/OracleClient.java

```diff
@@ -287,9 +290,7 @@ public PreparedStatement getPreparedStatement(Connection connection, String sql,
         PreparedStatement statement = connection.prepareStatement(sql);
         // This is a heuristic, not exact science. A better formula can perhaps be found with measurements.
         // Column count is not known for non-SELECT queries. Not setting fetch size for these.
-        if (columnCount.isPresent()) {
-            statement.setFetchSize(max(100_000 / columnCount.get(), 1_000));
-        }
+        statement.setFetchSize(jdbcFetchSize.orElseGet(() -> columnCount.map(cols -> max(100_000 / cols, 1_000)).orElse(0)));
```

Comment:
0 -> 1000 ?

Reply:
https://github.com/trinodb/trino/pull/22670#discussion_r1677783729

don't want to change current logic.